metadata/metadata.go

```diff
@@ -47,18 +51,29 @@ func FromContext(ctx context.Context) (Metadata, bool) {
 		return nil, ok
 	}
 
-	// capitalise all values
-	newMD := make(map[string]string)
+	newMD := make(Metadata)
+
 	for k, v := range md {
-		newMD[strings.Title(k)] = v
+		key, err := base32.StdEncoding.DecodeString(k)
+		if err != nil {
+			return nil, false
```

Comment:
Should you just continue? What if we're trying to be backwards compatible and there are no base32 encoded strings

Reply:
inside context we always have encoded metadata

Reply:
or you mean if we have older micro aff and never?

Reply:
push fix